src/mixins/SharedStoreMixin.js

```diff
@@ -0,0 +1,63 @@
+import { detach, types } from "mobx-state-tree";
+import Types from "../core/Types";
+
+const SharedStore = {};
+
+const SharedStoreModel = (childrenType) => types.model({
```

Comment:
What is the correlation between these set of SharedStore primitives and the set within SharedChoiceStore? Seems like these are basically the same, do we need both?

Reply:
Also, the SharedChoiceStore seems very specific to include directly into AnnotationStore, is this the intent? Seeing the factory based (and agnostic) SharedStore seems more flexible?

Reply:
so, `SharedStore` is more like a cache because we're creating a store within the `preProecssSnapshot` where there's no access to the tree. after that, when the node is created, the store is retrieved from the cache and attached to the tree

speaking of the AnnotationStore, it's done on purpose. attaching the store to the taxonomy itself will not solve the problem because every page has its own set of tags. the whole concept of shared storage doesn't work in this case

i had an idea of attaching it to the annotation, but it doesn't make sense as well – the list of choices can only change on the task level. keeping shared stores on the AnnotationStore level allows skipping the loading phase on other annotations

unfortunately, there's no other place between annotation and annotation store to fit the stores in

Reply:
Ok so we are keeping the concept of SharedChoiceStore, and that makes sense if its a global concept to have in the AnnotationStore. The one follow up to that which I was wondering, is where is this `[src/mixins/SharedStoreMixin.js](https://github.com/heartexlabs/label-studio-frontend/pull/935/files/7b441225e3466688e0e206f1c162021efeb573e4#diff-ea82889da1ec3e6f589dc270ac9d2aecf064ccf35b5cf6a298d8e464799a7dd9)` used? All I see that we are using is the [src/mixins/SharedChoiceStore/extender.js](https://github.com/heartexlabs/label-studio-frontend/pull/935/files/7b441225e3466688e0e206f1c162021efeb573e4#diff-bcd52178dea55c61de0ac21e27865af21f94f1980b98ed19723dc34df82ebe88)

Reply:
it's a mixin that is composed inside the [`AnnotationStore`](https://github.com/heartexlabs/label-studio-frontend/pull/935/files/7b441225e3466688e0e206f1c162021efeb573e4#diff-2aba458b602dc32bb13ce2d9e19b257d79a23f6eaab9e3d9903814b809066cd4R482-R485)

Reply:
That one I see as the canonical, but this mixin we are commenting in, its not found or used anywhere I can see? I searched for imports for `mixins/SharedStoreMixin` and I don't see this used at all.